middlewares/authorizeRoles.js

```diff
@@ -0,0 +1,57 @@
+const { ROLES } = require("../constants/users");
+
+const VALID_ROLES = [ROLES.SUPERUSER, ROLES.APPOWNER, ROLES.MEMBER];
+
+/**
+ * Check if the user has authorization based on their role.
+ * @param {Array} allowedRoles - Allowed roles for API consumption.
+ * @param {Object} userRoles - Roles information of the current user.
+ * @returns {Boolean} - Whether the current user is authorized or not.
+ */
+const userHasPermission = (allowedRoles, userRoles) => {
+  let permission = false;
+  Object.keys(userRoles).forEach((role) => {
+    if (userRoles[`${role}`]) {
+      if (allowedRoles.includes(role)) permission = true;
+    }
+  });
+
+  return permission;
+};
+
+/**
+ * Check if the user has authorization based on their role.
```

Comment:
Why userHasPermission and validateRoles have same description? Please be more specific

Reply:
Thanks for pointing this out.